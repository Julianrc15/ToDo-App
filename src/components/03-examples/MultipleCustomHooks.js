import React from 'react'
import { useCounter } from '../../hooks/useCounter'
import { useFetch } from '../../hooks/useFetch'
import '../01-useState/02-useEffect/effects.css'

export const MultipleCustomHooks = () => {

  const  {counter,  increment} = useCounter(1);

  const {loading,data} = useFetch (`https://www.breakingbadapi.com/api/quotes/${counter}`)
  // console.log(state);

  const {author,quote} =!!data && data[0];

  console.log(loading);


  return (

    <div>
    <h1>BreakingBad Qoutes</h1>
    <hr/>
    {
      loading
      ?
      (
        <div className="alert alert-info text-center">
            loading...
        </div>
      ) 
      :
      (

        <blockquote className="blockquote text-end">

          <p className="mb-3">{quote}</p>
          <footer className="blockquote-footer"> {author} </footer>

        </blockquote>

      )


    }

    <button 
    onClick={ increment }
    className='btn btn-primary'
    >
      Next quote
    </button>

    

    </div>
  )
}
