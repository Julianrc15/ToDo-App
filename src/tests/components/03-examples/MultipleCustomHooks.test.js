import React from 'react';
import {shallow}from 'enzyme';
import {MultipleCustomHooks} from '../../../components/03-examples/MultipleCustomHooks'
import { useFetch } from '../../../hooks/useFetch';
import { useCounter } from '../../../hooks/useCounter';

jest.mock('../../../hooks/useFetch')
jest.mock('../../../hooks/useCounter')




describe('Multiplespruebas en MultipleCustomHooks',()=>{

   
        beforeEach( () => {
            useCounter.mockReturnValue({
                counter: 10,
                increment: () => {}
            })
        })

    test('debe mostrase correctamente',()=>{

        useFetch.mockReturnValue({
            data:null, 
            loading:true, 
            error:null
        })

        const wrapper= shallow (<MultipleCustomHooks/>)

        expect(wrapper).toMatchSnapshot();

    })

    test('debe mostrar informaciÃ³n', ()=>{

        useFetch.mockReturnValue({
            data:[{
                author: 'John',
                quote:'hola Mundo'
            }], 
            
            loading:false, 
            error:null
        });
        const wrapper= shallow (<MultipleCustomHooks/>)

        expect(wrapper.find('.alert').exists() ).toBe(false);
        expect(wrapper.find('.mb-3').text().trim()).toBe('hola Mundo')
        expect(wrapper.find('footer').text().trim()).toBe('John')

        console.log(wrapper.html())

    })



})