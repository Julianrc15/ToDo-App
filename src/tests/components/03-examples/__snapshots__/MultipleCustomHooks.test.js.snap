// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Multiplespruebas en MultipleCustomHooks debe mostrase correctamente 1`] = `
<div>
  <h1>
    BreakingBad Qoutes
  </h1>
  <hr />
  <div
    className="alert alert-info text-center"
  >
    loading...
  </div>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Next quote
  </button>
</div>
`;
